src/transforms/groupby.js

```diff
@@ -152,13 +153,15 @@ function transformOne(trace, state) {
         styleLookup[styles[i].target] = styles[i].value;
     }
 
-    var newDataByGroup = {};
+    // An index to map group name --> expanded trace index
+    var groupIndex = {};
 
     for(i = 0; i < groupNames.length; i++) {
         groupName = groupNames[i];
+        groupIndex[groupName] = i;
```

Comment:
Naming is tricky. It's actually the array index that I'm indexing. So it should be called `indexIndex` :/